src/main/java/frc/robot/utilities/DebugEntry.java

```diff
@@ -27,39 +27,56 @@ public class DebugEntry<T> {
   public DebugEntry(T defaultValue, String name, Subsystem subsystem) {
     this.name = name;
     this.lastValue = defaultValue;
+
+    // if Double
     if (defaultValue instanceof Double) {
       localEntry = datalog.start("/" + subsystem.getName() + "/" + name, "double");
       localConsumer = (a) -> datalog.appendDouble(localEntry, (Double) a, 0);
+
+      // if String
     } else if (defaultValue instanceof String) {
       localEntry = datalog.start("/" + subsystem.getName() + "/" + name, "string");
       localConsumer = (a) -> datalog.appendString(localEntry, (String) a, 0);
+
+      // if Boolean
     } else if (defaultValue instanceof Boolean) {
       localEntry = datalog.start("/" + subsystem.getName() + "/" + name, "boolean");
       localConsumer = (a) -> datalog.appendBoolean(localEntry, (Boolean) a, 0);
+
+      // if Other
     } else {
       DriverStation.reportWarning("Unsupported data type.", false);
     }
-    if (localEntry != null) {
-      if (!Robot.isCompetition) {
-        networkTab = Shuffleboard.getTab(subsystem.getName());
 
-        if (!entries.containsKey(name)) {
-          networkEntry = networkTab.add(name, defaultValue).getEntry();
-          entries.put(name, networkEntry);
-        } else {
-          String errorMessage =
-              "Duplicate ShuffleboardEntry on " + networkTab.getTitle() + " tab: " + name;
-          DriverStation.reportError(errorMessage, false);
-        }
+    if (!Robot.isCompetition && localEntry != null) {
+      networkTab = Shuffleboard.getTab(subsystem.getName());
+
+      if (!entries.containsKey(name)) {
+        networkEntry = networkTab.add(name, defaultValue);
+        entries.put(name, networkEntry);
+      } else {
+        String errorMessage =
+            "Duplicate ShuffleboardEntry on " + networkTab.getTitle() + " tab: " + name;
+        DriverStation.reportError(errorMessage, false);
       }
-      localConsumer.accept(defaultValue);
     }
+    localConsumer.accept(defaultValue);
```

Comment:
This is why the check is failing. Needs to happen whether or not we are in competition mode, but skipped if the localEntry is null (i.e. wrong type was used)